src/vcpkg/commands.build.cpp

```diff
@@ -1127,53 +1131,57 @@ namespace vcpkg
                              " is disabled due to missing abi info for ",
                              dep_abi.key,
                              '\n');
-                return nullopt;
+                return;
             }
         }
 
         std::vector<AbiEntry> abi_tag_entries(dependency_abis.begin(), dependency_abis.end());
 
-        const auto& abi_info = action.abi_info.value_or_exit(VCPKG_LINE_INFO);
-        const auto& triplet_abi = paths.get_triplet_info(abi_info);
-        abi_tag_entries.emplace_back("triplet", triplet.canonical_name());
+        const auto& triplet_abi = paths.get_triplet_info(pre_build_info, toolset);
+        abi_info.triplet_abi.emplace(triplet_abi);
+        const auto& triplet_canonical_name = action.spec.triplet().canonical_name();
+        abi_tag_entries.emplace_back("triplet", triplet_canonical_name);
         abi_tag_entries.emplace_back("triplet_abi", triplet_abi);
-        abi_entries_from_abi_info(fs, grdk_cache, abi_info, abi_tag_entries);
+        auto& fs = paths.get_filesystem();
+        abi_entries_from_pre_build_info(fs, grdk_cache, pre_build_info, abi_tag_entries);
 
         // If there is an unusually large number of files in the port then
         // something suspicious is going on.
         constexpr int max_port_file_count = 100;
 
         std::string portfile_cmake_contents;
-        std::vector<Path> files;
-        std::vector<std::string> hashes;
         auto&& port_dir = action.source_control_file_and_location.value_or_exit(VCPKG_LINE_INFO).source_location;
-        Path abs_port_file;
-        auto port_files = fs.get_regular_files_recursive_lexically_proximate(port_dir, VCPKG_LINE_INFO);
-        if (port_files.size() > max_port_file_count)
+        auto raw_files = fs.get_regular_files_recursive_lexically_proximate(port_dir, VCPKG_LINE_INFO);
+        if (raw_files.size() > max_port_file_count)
         {
             msg::println_warning(
-                msgHashPortManyFiles, msg::package_name = action.spec.name(), msg::count = port_files.size());
+                msgHashPortManyFiles, msg::package_name = action.spec.name(), msg::count = raw_files.size());
         }
 
-        for (auto& port_file : port_files)
+        std::vector<Path> files;         // will be port_files without .DS_Store entries
+        std::vector<std::string> hashes; // will be corresponding hashes
+        for (auto& port_file : raw_files)
         {
             if (port_file.filename() == ".DS_Store")
             {
                 continue;
             }
-            abs_port_file = port_dir;
-            abs_port_file /= port_file;
 
+            files.push_back(port_dir / port_file);
+            const auto& abs_port_file = files.back();
```

Comment:
Can't you use `emplace_back` here so you already have the reference to the last element?

Reply:
If it works on our compiler support matrix, sure

Reply:
> If it works on our compiler support matrix, sure

It should work: https://en.cppreference.com/w/cpp/compiler_support/17#C.2B.2B17_library_features
Search for [P0084R2](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0084r2.pdf).

Reply:
Thanks :)